src/utils/internal.ts

```diff
@@ -548,6 +548,66 @@ export function getTicker(token: string | SecurityToken): string {
 /**
  * @hidden
  */
-export function calculateMoment(date: Date | undefined, context: Context): Moment | null {
-  return (date && dateToMoment(date, context)) ?? null;
+function secondsInUnit(unit: CalendarUnit): number {
+  const SECOND = 1;
+  const MINUTE = SECOND * 60;
+  const HOUR = MINUTE * 60;
+  const DAY = HOUR * 24;
+  const WEEK = DAY * 7;
+  const MONTH = DAY * 30;
+  const YEAR = DAY * 365;
+
+  switch (unit) {
+    case CalendarUnit.Second: {
+      return SECOND;
+    }
+    case CalendarUnit.Minute: {
+      return MINUTE;
+    }
+    case CalendarUnit.Hour: {
+      return HOUR;
+    }
+    case CalendarUnit.Day: {
+      return DAY;
+    }
+    case CalendarUnit.Week: {
+      return WEEK;
+    }
+    case CalendarUnit.Month: {
+      return MONTH;
+    }
+    case CalendarUnit.Year: {
+      return YEAR;
+    }
+  }
+}
+
+/**
+ * @hidden
+ * Transform a conversion util into a version that returns null if the input is falsy
+ */
+export function optionize<InputType, OutputType>(
+  converter: (input: InputType, context: Context) => OutputType
+): (val: InputType | null | undefined, context: Context) => OutputType | null {
+  return (value: InputType | null | undefined, context: Context): OutputType | null => {
+    value = value ?? null;
```

Comment:
Avoid assigning values to function parameters. Use a different variable for this